web/src/ui_util.ts

```diff
@@ -22,14 +22,39 @@ export function place_caret_at_end(el: HTMLElement): void {
     }
 }
 
-export function replace_emoji_with_text($element: JQuery): void {
+export function replace_emoji_name_with_unicode_emoji_code($element: JQuery): void {
     $element
         .find(".emoji")
         .text(function () {
+            // We don't have corresponding unicode for a custom emoji.
             if ($(this).is("img")) {
                 return $(this).attr("alt") ?? "";
             }
-            return $(this).text();
+
+            // Emojis have a class with the emoji code next to them e.g. ".emoji-1f419".
+            // The code "1f419" represents an octopus.
+            const emoji_class_with_code: string | undefined = $(this).attr("class");
+            if (emoji_class_with_code === undefined) {
+                return $(this).text();
+            }
+
+            const regex = /emoji-(\w+)/;
+            const match = regex.exec(emoji_class_with_code);
+            const emoji_code = match?.[1] ?? "";
+
+            const hex_emoji_code = Number.parseInt(emoji_code, 16);
```

Comment:
I think these variable names are quite confusing. The string you get should be called `emoji_code_hex_string`, because it's a hexadecimal string. The thing you compute here is just an integer, so would most properly be called something like `emoji_code` or `emoji_code_int`.